src/site/stages/build/process-cms-exports/index.js

```diff
@@ -10,7 +10,140 @@ const {
+    //
+    // If it isn't handled in the transformer, the post-transformation
+    // validation will fail because of a circular reference (AJV will throw
+    // up).
+    //
+    // If the modified child isn't accounted for in the transformed schema, it
+    // won't be valid (assuming we've omited a normally-required property to
+    // avoid the circular reference).
+    return a;
+  }
+  return false;
+};
+
+/**
+ * @param {Object} entity - The entity we're validating
+ * @throws {Error} If the entity is invalid
+ */
+const validateInput = entity => {
+  // Pre-transformation JSON schema validation
+  const rawErrors = validateRawEntity(entity);
+  if (rawErrors.length) {
+    /* eslint-disable no-console */
+    console.warn(
+      chalk.yellow(
+        `${toId(entity)} (${getContentModelType(
+          entity,
+        )}) is invalid before transformation:`,
+      ),
+    );
+    console.warn(`${rawErrors.map(e => JSON.stringify(e, null, 2))}`);
+    rawErrors.forEach(e => {
+      console.warn(
+        chalk.yellow(`Data found at ${e.dataPath}:`),
+        JSON.stringify(get(entity, e.dataPath.slice(1))),
+      );
+    });
+    console.warn(`-------------------`);
+    /* eslint-enable no-console */
+
+    // Abort! (We may want to change this later)
+    throw new Error(`${toId(entity)} is invalid before transformation`);
+  }
+};
+
+/**
+ * @param {Object} transformedEntity - The entity after transformation
+ * @throws {Error} If the entity is invalid
+ */
+const validateOutput = (entity, transformedEntity) => {
+  const transformedErrors = validateTransformedEntity(transformedEntity);
+  if (transformedErrors.length) {
+    /* eslint-disable no-console */
+    console.warn(
+      chalk.yellow(
+        `${toId(entity)} (${getContentModelType(
+          entity,
+        )}) is invalid after transformation:`,
+      ),
+    );
+    console.warn(`${transformedErrors.map(e => JSON.stringify(e, null, 2))}`);
+    transformedErrors.forEach(e => {
+      console.warn(
+        chalk.yellow(`Data found at ${e.dataPath}:`),
+        JSON.stringify(get(transformedEntity, e.dataPath.slice(1))),
+      );
+    });
+    console.warn(`-------------------`);
+    /* eslint-enable no-console */
+
+    // Abort! (We may want to change this later)
+    throw new Error(`${toId(entity)} is invalid after transformation`);
+  }
+};
+
 const entityAssemblerFactory = contentDir => {
+  /**
+   * @param {Object} entity - The entity with entity references
+   * @param {Ancestor[]} ancestors - A list of all ancestors
+   * @param {function} assembleTree - The assembleEntityTree closure; defined as
+   *                                  a parameter here because eslint didn't
+   *                                  like using it before it was defined
+   * @return {Object} The entity with the references filled in
+   */
+  const expandEntityReferences = (entity, ancestors, assembleTree) => {
+    const filteredEntity = getFilteredEntity(entity);
```

Comment:
And then it looks like we filter it again.  Do we need to `getFilteredEntity` here?